packets/game_state_packet.cpp

```diff
@@ -0,0 +1,87 @@
+#include "../include/game_state_packet.h"
+#include <cstring>
+
+void GameStatePacket::serialize(const GameStatePacket& packet, char*& outData) {
```

Comment:
Is it the caller's responsibility to ensure outData has enough memory to store the serialized packet? I think that's fine if so, but it may be good to note it somewhere. (This goes for `InputPacket::serialize` as well)